### ISSUE TITLE
Create new cargo release to address safety issue

### ISSUE THREAD

**Issue by ammaraskar** [slide-rs/atom#13]
I noticed there's been a bunch of changes on master contributed by @dbkaplun that haven't made their way out onto the cargo crate. Most importantly, the latest release, 0.3.5 is missing https://github.com/slide-rs/atom/commit/dc096ef61a7cf69ce1a833a277b1c2a7c726c129 which fixes a fairly important soundness issue around the `Send` trait's bounds.

Without this commit, it's possible to `Send` across unsafe objects like `Rc` and potentially run into data races like so:

```rust
#![forbid(unsafe_code)]

extern crate atom;

use atom::Atom;
use std::rc::Rc;
use std::sync::mpsc;
use std::{thread, time, mem};

const NUM_CLONES: usize = 10000000;

fn main() {
    let x = Rc::new(());
    let y = Rc::clone(&x);

    // Place our Rc in the Atom through a Box.
    let shared_atom = Atom::empty();
    shared_atom.swap(Box::new(x));

    let child = thread::spawn(move || {
        // We now have the Rc shared across a thread.
        let x = shared_atom.take().unwrap();
        for _ in 0..NUM_CLONES {
            let clone = x.clone();
            mem::forget(clone);
        }
    });

    for _ in 0..NUM_CLONES {
        let clone = y.clone();
        mem::forget(clone);
    }

    // Wait for the spawned thread to finish its cloning.
    child.join().unwrap();

    // We made NUM_CLONES on both threads plus 2 references in the main thread.
    // But in reality we'll see that the strong_count varies across every run.
    assert_eq!(Rc::strong_count(&y), (NUM_CLONES * 2) + 2);
}

```

This could lead to use-after-frees and is a fairly easy mistake to make.

**Comment by torkleyy** [comment]
Thanks for the report!

I released a patch, 0.3.6 which contains a backport of the linked commit (https://github.com/slide-rs/atom/tree/v0.3).

I also published the current master as 0.4.

**Comment by ammaraskar** [comment]
Thank you for addressing this! :)
